lib/fetch/index.js

```diff
@@ -1562,78 +1556,155 @@ function httpNetworkFetch (
 
           context.on('terminated', onResponseAborted)
 
+          const codings =
+            headers
+              .get('content-encoding')
+              ?.toLowerCase()
+              .split(',')
+              .map((x) => x.trim()) ?? []
+
+          const decoders = []
+
+          // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
+          for (const coding of codings) {
+            if (/(x-)?gzip/.test(coding)) {
+              decoders.push(zlib.createGunzip())
+            } else if (/(x-)?deflate/.test(coding)) {
+              decoders.push(zlib.createInflate())
+            } else if (coding === 'br') {
+              decoders.push(zlib.createBrotliDecompress())
+            } else {
+              // TODO: What to do when coding is invalid or unsupported?
```

Comment:
Spec says to [just passthrough](https://fetch.spec.whatwg.org/#handle-content-codings). Similar to the `identity` encoding.

This means any unknown non-`identity` encoding should invalidate this decoder chain.

Reply:
Would you mind opening a PR to fix?

Reply:
Fixed in https://github.com/nodejs/undici/commit/963ca033466cacbf6c72e299871f1e6aa4d1ced8